### DISCUSSION TITLE
`useButton` and `useOverlayTriggerState` interacting in strange way

### DISCUSSION THREAD

**Discussion by brandonpittman** [adobe/react-spectrum#3593] [Q&A]
I have a few overlays in an app I'm working on. One of them is a little action menu. Everything behaves as expected.

https://user-images.githubusercontent.com/967145/193762314-790befec-df2a-49c3-868a-c03b318f806a.mp4

This date picker, however, can't close when clicking the trigger a second time.

https://user-images.githubusercontent.com/967145/193762483-cf4b750a-a45c-46bd-8388-21bc0b069973.mp4

More accurately, it sets `state.isOpen` to `false` as expected, and then sets it back to `true` again. I noticed that if I click without releasing, and drag the mouse away, it stays closed. The state only changes back to `true` on `pointerUp`. I've confirmed the `press` event from `useButton`'s `buttonProps` is only firing once though.

Any idea what could be happening here?

**Comment by LFDanLu** [comment]
Apologies about the delay, do you happen to have a codesandbox you can share for your datepicker? I assume the trigger button is getting its props from useDatePicker?
